### ISSUE TITLE
1992 Called. It wants your homepage design

### ISSUE THREAD

**Issue by activestylus** [urweb/urweb#95]
Seriously, this project looks promising but 90% of people visiting your site will turn right back around and never come back. The logo and overall layout looks like it was designed by colorblind math professors. Have yall considered getting a professional site/logo done? With a proper bit of marketing you might finally get a bit more momentum/mindshare.

Just a thought..

Aside from that, really nice work. I'm not hip to the syntax but I like how integrated and performant this solution is and I wish you all success.

**Comment by achlipala-biz** [comment]
You're not the first to make this suggestion.  FWIW, I _am_ color-blind, and I'm almost a math professor.

I've tried before to find a volunteer to take over running the whole web site, actually.  No one stepped up, though someone suggested a new design template, where the mailing-list participants were split on whether it was an improvement.

**Comment by saltvedt** [comment]
Hey! Glad I'm not the only one 😃 

See #54 for my proposal, based on feedback on the (in my opinion) prettier _original_ proposal below.



![urweb_main fw](https://user-images.githubusercontent.com/1760730/31890152-70fbe726-b802-11e7-84a9-3ad89c26473f.png)

**Comment by activestylus** [comment]
@saltvedt That is already MUCH better. There's still room for improvement but it is mind-boggling to me that anyone could look at that and NOT see it as an step up from what is currently live.

**Comment by activestylus** [comment]
@achlipala-biz No offense intended, although my proximity to the mark does have me laughing! We all have our strengths no? :)

**Comment by activestylus** [comment]
The current site is fine if you are from a computer science background, but there's an increasing mindshare of fullstack developers/designers who will 100% judge a language/stack/frameowrk based on its brand. Yes it sounds superficial, but having a decent online presence can only help, though I would concur that tutorials, guides and blog posts are definitely top priority

**Comment by saltvedt** [comment]
@karsar If I stumble upon a website with designs like the current site, my first thought will be that the project is old and no longer actively maintained. Even if potential users realises that the project is alive, the lacklustre design sends a very clear signal that design is unimportant to Ur/Web. But web design is extremely important in web development. 

What is happening right now is equivalent to presenting a Commodore 64 and expecting the user to read the manual to figure out that they infact are dealing with a modern, high-end computer.

Like it or not, web frameworks, like everything else, is judged by appearances. And those first impressions matter.

If Ur/Web has growth ambitions, a renewed focus on design and branding is necessary to be successful. And obviously through the nature of open source projects, better documentation and other improvements can through a larger user base.

**Comment by ansarizafar** [comment]
I totally agree with @saltvedt and @activestylus.

**Comment by ashalkhakov** [comment]
Have you seen what other devs are doing? Devs: "We just want our job to be easier!" Framework authors: "Our framework makes things EASY! Try it!" So every framework or a library that gains traction is first and foremost easy to begin with.

A decent, modern website design would be help to sell the idea to developers.

> The current site is fine if you are from a computer science background, but there's an increasing mindshare of fullstack developers/designers who will 100% judge a language/stack/frameowrk based on its brand.

Yes, and also we developers like to think we are objective and impassionate, but we are not. By the way... where's the next bandwagon?

**Comment by fullreset** [comment]
@achlipala-biz I am willing to create a new site, but I cannot promise I will or can do so quickly -- as I have other obligations. But ... should we talk about this? I can also host the site and you can redirect as a CNAME; or I send can you the code.

**Comment by achlipala** [comment]
Thanks for volunteering, @fullreset!  I like the model you're proposing, though I'd hope to avoid a CNAME and point the proper hostname directly to your IP address.  Probably the most complex aspect of hosting is running Ur/Web application demos.

I have to think about what qualifications a potential web-site maintainer should have.  Can you remind me when you first started participating in the Ur/Web community?  What have you done and what are you doing with Ur/Web?

P.S.: I accidentally posted with the wrong GitHub account before.  The one I'm using here is the right one for Ur/Web business.

**Comment by fullreset** [comment]
I've only recently started using Ur/Web in my work -- call it a few weeks. :) But I have several decades of experience in software development, in fields ranging 3D and computer vision to compilers to bank software. So I do sort of know what I'm doing. :P

And I'm currently working with Ur/Web to build ERP software and a few personal projects.

PS Directly pointing to my server IP would not be a problem; but I could also just send you the code to host on `impredicative.com` -- since you apparently already have a server. Works either way.

**Comment by ansarizafar** [comment]
Proper website with modern UI is not enough, Beginner friendly documentation/Manual (Gitbook) in simple language and use of community building/support tools (Google groups /Gitter / Stackoverflow) is also required.

**Comment by achlipala** [comment]
I think it might be doing the world a disservice to have an Ur/Web manual in "simple language," since serious cognitive sophistication is required to use the language.

**Comment by ansarizafar** [comment]
Ur/Web documentation/manual in simple language is the first requirement If we want to make UrWeb a mainstream language for web development.

**Comment by achlipala** [comment]
Oh, I don't want to make Ur/Web a mainstream language, any more than Haskell is mainstream.

**Comment by ansarizafar** [comment]
Ok. Got it.

**Comment by fullreset** [comment]
I don't think the node.js people are going to come running; but a somewhat larger community would be useful -- shared knowledge, resources, etc. A simple example:  I added basic inline SVG the other day, which of course was easy; but I don't imagine I'm the first one to do this -- more sharing would be nice. 

@ansarizafar I agree that more docs, examples, demos, etc. would help; but I also think that a better organized home page for the project would help a bit. (It wouldn't need to be fancy, just cleaner and more comprehensive -- and more up to date.)

**Comment by ansarizafar** [comment]
@fullreset I totally agree. Right now it's very difficult to find a solution If u stuck.

**Comment by fullreset** [comment]
@ansarizafar Maybe we solve this problem ourselves. I put two tiny things here https://github.com/fullreset/urweb-utils.git, and then we can make or gather more. :)

**Comment by ashalkhakov** [comment]
@fullreset awesome! Would you kindly publish it on NPM?

I guess a separate discussion is in order. How to distribute Ur/Web packages? Should we move it to the mailing list?

**Comment by ansarizafar** [comment]
@fullreset A lot needs to be done to build a community and an echo system around UrWeb.

Other projects are using Github issues, Google groups, discord or http://gitter.im/ for discussions/community support. I can manage any selected channel for Urweb.

**Comment by fullreset** [comment]
@ansarizafar I don't see a need for NPM -- `git clone` does the job, we just need a place to "announce" or refer to all the resources found or created. It could even just be another git repository that serves as an index. (If the collection grows we can find another solution.)

I haven't used gitter.im, but I'll take a look. And Google things I'd rather not use because I'm behind a really big firewall. :P

**Comment by ansarizafar** [comment]
We can use a  simple Github repo like this one https://github.com/sindresorhus/awesome-nodejs  to collect resources related to UrWeb. Gitter is used by all major companies like Google. Miucrosoft, Airb&b, Amazone etc. https://discordapp.com/ is also a free and used by many projects. These tools are great for community support. I have many questions related to UrWeb but no place to ask these questions.

**Comment by achlipala** [comment]
My position is that the mailing list and IRC channel are already available for asking questions and getting quick answers.

**Comment by ansarizafar** [comment]
In my opinion Google groups, Gitter, Discord etc. are much better options with many modern features.

**Comment by ansarizafar** [comment]
Check this https://github.com/facebook/docusaurus

**Comment by fullreset** [comment]
A random thought: 

An expanded version of Professor Achlipala's tutorials 1 & 2 in the style of [Learn X in Y Minutes (Standard ML link -- since applicable)](https://learnxinyminutes.com/docs/standard-ml/) -- hosted on impredicative.com as well as this site -- showing edge cases and explaining small syntactic details that take a while to figure out might be helpful in increasing interest in this cool Ur/Web thing.

**Comment by 0joshuaolson1** [comment]
I understand wanting a 'trendy' and more readable website, but 'modern' webdev often leaves a bad impression when it comes to:
-  limited data plans/devices/loading time/battery (useless images/CSS/animations, reader has other tabs/windows open in a crappy browser)
- accessible scrolling/pagination
- 'text' that's actually images

**Comment by achlipala** [comment]
Apologies in advance for any violation of GitHub etiquette, but....

@shiftreset, especially if you are still up for maintaining an Ur/Web community web server/site, would you mind contacting me by e-mail to continue the conversation there?  I'm `adamc@csail.mit.edu`, and I couldn't find your e-mail address or other identity information here.  Thank you.

**Comment by fullreset** [comment]
@achlipala yes, i would still be interested in doing this. i will email you in the next day or two ...

**Comment by wisn** [comment]
Any update for this?

**Comment by achlipala** [comment]
No one followed up on things the last time around.  The state, in my mind, is I'd like to find a volunteer to run a "showcase" web site for Ur/Web, which doesn't need to be particularly trusted anymore, since the official versions of the code, etc., are all on GitHub now.  Whoever runs the showcase site could install a design that satisfies the community as sufficiently newfangled-looking, etc.  It would be good to have Ur/Web application demos running there, but otherwise probably it just calls for static content.

**Comment by wisn** [comment]
@achlipala Which repository is it? I will give a try to build one with the newest clean web design

**Comment by achlipala** [comment]
Actually, I mischaracterized a bit: it's the whole `urweb` organization on GitHub that is now the source of software ground truth, though the `urweb` repository within it is by far the most important one.

Thanks for volunteering to create a better web design!  I believe there was actually pretty broad satisfaction with the one someone proposed before, but it doesn't hurt to suggest something new.  I suggest using our mailing list for discussion about a proposed design.  Really, though, I expect the hosting of a site with running demo apps to be the more intensive part.

**Comment by saltvedt** [comment]
My previous proposal is in #54 

It uses GitHub pages and Jekyll. I forked urweb and set up a branch here: https://github.com/saltvedt/urweb/tree/gh-pages

GitHub magic then makes it so that the website is accessible here: http://saltvedt.net/urweb/

See GitHub documentation here: https://help.github.com/en/articles/configuring-a-publishing-source-for-github-pages
